### PR TITLE
[#49] drain.sh script should catch the scaledown status and return the error code

### LINKED ISSUE #49: drain.sh script should catch the scaledown status and return the error code
**Describe the bug**
The drain.sh script didn't check the response code from the scaledown call and if the scaledown fails the drainer pod
exits without error, this causes the operator to delete the PVC and may lose messages in it.

**To Reproduce**
1. Get X version of the source
2. '...'

**Platform (please complete the following information):**
 - OS: [e.g. Fedora]
 - Version [e.g. 32]
 - CEKit version [e.g. 3.5.0]
 - Container build tool [e.g. podman]

**Expected behavior**
Optional, use if not obvious. A clear and concise description of what you expected to happen.

**Screenshots**
Optional, add screenshots to help explain your problem.

**Additional context**
Optional. Add any other context about the problem here.

### PR DESCRIPTION
**Pull request by gaohoward**

### CODE REVIEW

**Comment by brusdev** [comment]
The fix LGTM but I don't see any test, are you planning to add a test in the activemq-artemis-operator code base?

**Comment by gaohoward** [comment]
> The fix LGTM but I don't see any test, are you planning to add a test in the activemq-artemis-operator code base?

At the moment I don't have an idea how to simulate a scaledown failure so that'll be a task to do (I'll create an upstream issue on operator side).  However the test I'm updating in this PR https://github.com/artemiscloud/activemq-artemis-operator/pull/527 will catch the current scaledown failure (meaning it fails on main until the scaledown issue has been fixed).

File: modules/activemq-artemis-launch/added/drain.sh [hunk 1]
```diff
@@ -71,4 +110,17 @@ else
...
+HTTP_CODE=`echo $RET_CODE | python2 -c "import sys, json; print json.load(sys.stdin)['status']"`
+
+echo "curl return code ${HTTP_CODE}"
+
+if [ "${HTTP_CODE}" != "200" ]; then
+  echo "scaleDown is not successful, response: $RET_CODE sleep for debug"
+  exit 1
+fi
+
+echo "scaledown is successful sleep for debug"
```
Comment (brusdev) [inline modules/activemq-artemis-launch/added/drain.sh]: I don't see any sleep command, why `sleep for debug`?

Comment (rvais) [reply modules/activemq-artemis-launch/added/drain.sh]: Probably so the container stays up ...?

File: modules/activemq-artemis-launch/added/drain.sh [hunk 2]
```diff
@@ -26,31 +27,69 @@ function waitForJolokia() {
...
   if [ $? -ne 0 ]; then
     echo "Can't find ip to scale down to tried ${count} ips"
-    exit
+    exit 1
   fi
-
   echo "got ip ${ip} broker ip is ${BROKER_IP}"
   if [ "$ip" != "$BROKER_IP" ]; then
+    # found one, now get pod name
+    podName=$(echo $ENDPOINTS | python2 -c "import sys, json; print json.load(sys.stdin)['subsets'][0]['addresses'][${count}]['targetRef']['name']")
```
Comment (rvais) [inline modules/activemq-artemis-launch/added/drain.sh]: Shouldn't we rather use python3 these days?

File: modules/activemq-artemis-launch/added/drain.sh [hunk 3]
```diff
@@ -26,31 +27,69 @@ function waitForJolokia() {
...
   ip=$(echo $ENDPOINTS | python2 -c "import sys, json; print json.load(sys.stdin)['subsets'][0]['addresses'][${count}]['ip']")
   if [ $? -ne 0 ]; then
     echo "Can't find ip to scale down to tried ${count} ips"
-    exit
+    exit 1
   fi
-
   echo "got ip ${ip} broker ip is ${BROKER_IP}"
   if [ "$ip" != "$BROKER_IP" ]; then
+    # found one, now get pod name
```
Comment (rvais) [inline modules/activemq-artemis-launch/added/drain.sh]: the comment is opposite of the condition above. Shouldn't it be something like "Haven't found one, let's try through pod name" ?

**Comment by rvais** [comment]
I think we should use the service DNS name by default instead of IP addresses. At least that's what we use for clients in our tests. If service doesn't exist, or doesn't work, I think we got bigger problems than a scale down and the scale down should be aborted.

File: modules/activemq-artemis-launch/added/drain.sh (hunk 2 above)
Comment (gaohoward) [reply modules/activemq-artemis-launch/added/drain.sh]: well yes we should, I'll create a task. thx Roman.

File: modules/activemq-artemis-launch/added/drain.sh (hunk 3 above)
Comment (gaohoward) [reply modules/activemq-artemis-launch/added/drain.sh]: No the comment is correct. The aim here is to find a living pod that is not the "dead one". 
But maybe there is a problem with BROKER_IP comparison... (it is got from hostname -f which shouldn't be an ip address...)

Comment (rvais) [reply modules/activemq-artemis-launch/added/drain.sh]: So there should be maybe another comment, that would reflect that. This way the condition looks like 
"I got an IP address, but that IP address must not be IP address of broker" (in another words, I am searching for IP of anything else but broker). It's a bit confusing. Naming convention for variables might be at fault here, so comment could clarify that.

Comment (gaohoward) [reply modules/activemq-artemis-launch/added/drain.sh]: yeah let me make it more clear. thanks

**Comment by gaohoward** [comment]
> I think we should use the service DNS name by default instead of IP addresses. At least that's what we use for clients in our tests. If service doesn't exist, or doesn't work, I think we got bigger problems than a scale down and the scale down should be aborted.

The purpose of drain.sh is to find a 'live' pod to migrate messages to. So it looking through the pods using the headless service we exposed and find the first available, then it issues a mgmt request for scaledown.

File: modules/activemq-artemis-launch/added/drain.sh (hunk 1 above)
Comment (gaohoward) [reply modules/activemq-artemis-launch/added/drain.sh]: yes that's something I forgot to remove. :)

File: modules/activemq-artemis-launch/added/drain.sh (hunk 2 above)
Comment (gaohoward) [reply modules/activemq-artemis-launch/added/drain.sh]: this 
 if [ "$ip" != "$BROKER_IP" ]; then
is not right, the $BROKER_IP is indeed the pod's host name so the comparison always yields true. It should compare the podName with $BROKER_IP.

**Review by brusdev — APPROVED** [review]

**Pull request merged**